### PR TITLE
fix vae dtype when accelerate config using --mixed_precision="fp16"

### PR DESCRIPTION
**Pull request by xduzhangjiayu**
# What does this PR do?
Issue https://github.com/huggingface/diffusers/issues/9560
Fix vae dtype when accelerate config using --mixed_precision="fp16".

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
@sayakpaul

### CODE REVIEW

File: examples/controlnet/train_controlnet_sd3.py [hunk 1]
```diff
@@ -1094,7 +1094,7 @@ def load_model_hook(models, input_dir):
         weight_dtype = torch.bfloat16
 
     # Move vae, transformer and text_encoder to device and cast to weight_dtype
-    vae.to(accelerator.device, dtype=torch.float32)
+    vae.to(accelerator.device, dtype=weight_dtype)
```
Comment (sayakpaul) [inline examples/controlnet/train_controlnet_sd3.py:1097]: I think it should be like:

```python
if args.upcast_vae:
    vae.to(accelerator.device, dtype=torch.float32)
else:
    vae.to(accelerator.device, dtype=weight_dtype)
```

With `upcast_vae` being added to the arguments?

**Review by sayakpaul — COMMENTED** [review]
Thanks! Just left a comment.

**Comment by xduzhangjiayu** [comment]
Thanks for the comment, I have modified the code.

**Review by sayakpaul — APPROVED** [review]
Thanks!

**Pull request merged**